File: EqAspNet4Demo/Controllers/EasyQuery/AdvancedSearchController.cs
using System;
using System.Configuration;
using System.Web.Http;
using System.Data.SqlClient;

using Korzh.EasyQuery.Services;
using Korzh.EasyQuery.AspNet;

namespace EqAspNet4Demo.Controllers
{
    [RoutePrefix("api/easyquery")]
    public class AdvancedSearchController : EasyQueryApiController
    {
        protected override void ConfigureEasyQueryOptions(EasyQueryOptions options)
        {
            options.UseManager<EasyQueryManagerSql>();
            options.DefaultModelId = "NWindSQL";
            options.BuildQueryOnSync = true;

            ConnectionStringSettings connectionSettings = ConfigurationManager.ConnectionStrings["DefaultConnection"];
            if (connectionSettings == null || string.IsNullOrEmpty(connectionSettings.ConnectionString)) {
                throw new Exception("Fatal error: missing connecting string in web.config file");
            }

            options.UseDbConnection<SqlConnection>(connectionSettings.ConnectionString);

            var path = System.Web.Hosting.HostingEnvironment.MapPath("~/App_Data");
            options.UseModelLoader((_) => new FileModelLoader(path));
            options.UseQueryStore((_) => new FileQueryStore(path));

            // Uncomment this line if you want to load model directly from connection 
            // Do not forget to uncomment SqlClientGate registration in WebApiConfig.cs file
            //options.UseDbConnectionModelLoader(config => {
            //    // Ignores Asp.Net Identity tables
            //    config.AddTableFilter((table) => !table.Name.StartsWith("Asp") 
            //                                  && table.Name != "IdentityUsers"
            //                                  && table.Name != "__MigrationHistory");

            //    // Ignores Reports table
            //    config.AddTableFilter((table) => table.Name != "Reports");
            //});

            options.UsePaging(30);
        }
    }
}

File: EqWebFormsDemoVb/Controllers/AdvancedSearchController.vb
Imports System
Imports System.Configuration
Imports System.Data.SqlClient
Imports System.Web.Http

Imports Korzh.EasyQuery.AspNet
Imports Korzh.EasyQuery.Services

<RoutePrefix("api/easyquery")>
Public Class AdvancedSearchController : Inherits EasyQueryApiController

    Protected Overrides Sub ConfigureEasyQueryOptions(options As EasyQueryOptions)

        options.UseManager(Of EasyQueryManagerSql)()

        options.DefaultModelId = "NWindSQL"
        options.BuildQueryOnSync = True

        Dim connectionString As String = ConfigurationManager.ConnectionStrings.Item("DefaultConnection").ConnectionString
        options.UseDbConnection(Of SqlConnection)(connectionString)

        Dim path As String = System.Web.Hosting.HostingEnvironment.MapPath("~/App_Data")
        Dim modelLoader As Func(Of IServiceProvider, IModelLoader) =
                Function()
                    Return New FileModelLoader(path)
                End Function
        Dim queryStore As Func(Of IServiceProvider, IQueryStore) =
                Function()
                    Return New FileQueryStore(path)
                End Function
        options.UseModelLoader(modelLoader)
        options.UseQueryStore(queryStore)

        options.UsePaging(30)

    End Sub
End Class
